libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -775,13 +782,33 @@ class MapboxNavigation @VisibleForTesting internal constructor(
                 RoutesExtra.ROUTES_UPDATE_REASON_NEW
             }
         }
-        directionsSession.setRoutes(
+        internalSetNavigationRoutes(
             routes,
             initialLegIndex,
             reason,
         )
     }
 
+    private fun internalSetNavigationRoutes(
+        routes: List<NavigationRoute>,
+        legIndex: Int = 0,
+        @RoutesExtra.RoutesUpdateReason reason: String,
+    ) {
+        Log.e("lp_test", "internalSetNavigationRoutes")
+        threadController.getMainScopeAndRootJob().scope.launch(Dispatchers.Main.immediate) {
+            routeUpdateMutex.withLock {
```

Comment:
I tried explaining this in `RouteAlternativesController#paused`, could you take a look and let me know whether that's enough or not?

Reply:
yep, it's clear 👍 

IMO it's fine that users receive alternatives for the current location, we might mention it in docs. 

Additionally, sdk might skip an alternative that NN proposes and it's not in the original list.

Reply:
Sorry, I don't follow the last two points, could you rephrase?

Reply:
> That's not ideal though because the integration pattern is to reset anything that you get in NavigationRouteAlternativesObserver back to MapboxNavigation#setRoutes, so the RoutesObserver call will effectively be duplicated. 

got it 👍 That makes sense to me